File: medium/cpp/1947.cpp
// https://leetcode.com/problems/maximum-compatibility-score-sum/

class Solution
{
public:
    int maxCompatibilitySum(std::vector<std::vector<int>>& students, std::vector<std::vector<int>>& mentors)
    {
        const auto n = std::size(students);
        
        std::vector<int> idx(n);
        
        for (auto i = 0; i < n; i++)
        {
            idx[i] = i;
        }
        
        auto ans = 0;
        
        do
        {
            auto sum = 0;
            
            for (auto i = 0; i < n; i++)
            {
                for (auto j = 0; j < std::size(students[idx[i]]); j++)
                {
                    if (students[idx[i]][j] == mentors[i][j])
                    {
                        sum++;
                    }
                }
            }
            
            ans = std::max(ans, sum);
        }
        while (std::next_permutation(std::begin(idx), std::end(idx)));
        
        return ans;
    }
};

File: medium/python/1947.py
// https://leetcode.com/problems/maximum-compatibility-score-sum/description/

class Solution:
    def maxCompatibilitySum(self, students: List[List[int]], mentors: List[List[int]]) -> int:
        m = len(students)
        score = 0

        for perm in permutations(range(0, m)):
            cur = 0
            for i in range(0, m):
                cur += sum(x == y for x, y in zip(students[i], mentors[perm[i]]))
            score = max(score, cur)

        return score